apps/dash-mapd-demo/app.py

```diff
@@ -0,0 +1,737 @@
+
+    :return: city delay scatter graph.
+    """
+    start_f = f"{start} 00:00:00"
+    end_f = f"{end} 00:00:00"
+
+    days = ["Mon", "Tues", "Wed", "Thu", "Fri", "Sat", "Sun"]
+
+    count_df = []
+    state_f = state_select if len(state_select) > 0 else "NY"
+    title = (
+        f"Avg Arrival Delays By Cities for State <b>{state_f}</b>"
+        if dd_select == "arr"
+        else f"Avg Departure Delays By Cities for State <b>{state_f}</b>"
+    )
+    state_col = "origin_state" if dd_select == "arr" else "dest_state"
+    city_col = "origin_city" if dd_select == "arr" else "dest_city"
+
+    for i, day in enumerate(days):
+        count_query = (
+            f"SELECT AVG(arrdelay) AS {day}, {city_col} AS city FROM {table} WHERE {state_col} ='{state_f}' AND flight_dayofweek = {i + 1}"
+            f" AND {dd_select}_timestamp BETWEEN '{start_f}' AND '{end_f}' group by {city_col}"
+        )
+
+        try:
+            df_city_count = pd.read_sql(count_query, db_connect()).set_index("city")
+        except Exception as e:
+            print("Error reading count queries", e)
+            return {}
+
+        count_df.append(df_city_count)
+
+    count_df = pd.concat(count_df, axis=1, sort=True)
+
+    data = []
+    for city in count_df.index:
+        customdata = list(city for _ in range(7))
+        trace = go.Scatter(
+            y=days,
+            x=count_df.loc[city, :],
+            name=city,
+            customdata=customdata,
+            mode="markers",
+        )
+        data.append(trace)
+
+    layout = dict(
+        title=dict(
+            text=title,
+            font=dict(family="Open Sans, sans-serif", size=15, color="#515151"),
+        ),
+        automargin=True,
+        font=dict(family="Open Sans, sans-serif", size=13),
+        xaxis=dict(
+            title=dict(text="Minutes", font=dict(size=12)), zerolinecolor="#999999"
+        ),
+        hovermode="closest",
+        clickmode="event+select",
+        dragmode="select",
+        showlegend=True,
+    )
+
+    return {"data": data, "layout": layout}
+
+
+def query_helper(state_query, dd_select, start, end, weekday_query):
+    con = db_connect()
+    add_and = ""
+    if state_query:
+        add_and = "AND"
+    query = (
+        f"SELECT uniquecarrier AS carrier, flightnum, dep_timestamp, arr_timestamp, origin_city, dest_city "
+        f"FROM {table} WHERE {state_query} {add_and} {dd_select}_timestamp BETWEEN '{start}' AND '{end}' {weekday_query} limit 100"
+    )
+
+    try:
+        dff = pd.read_sql(query, con)
+        dff["flightnum"] = dff["carrier"] + dff["flightnum"].map(str)
+        dff.drop(["carrier"], axis=1)
+        return dff.to_dict("rows")
+    except Exception as e:
+        print(f"Error querying {query}", e)
+        raise PreventUpdate
+
+
+app.layout = html.Div(
+    className="container scalable",
+    children=[
+        html.Div(
+            id="banner",
+            className="banner",
```

Comment:
I'm not sure if this is intentional, but theres margin on the left and right side of the banner and the mockup seems to have none

Reply:
💯 Removed.